frontend/public/components/monitoring/alert-manager-yaml-editor.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+
+import {Base64} from 'js-base64';
+import {k8sPatch, K8sResourceKind} from '../../module/k8s';
+import {SecretModel} from '../../models';
+import {ButtonBar, history, LoadingBox, StatusBox} from '../utils';
+import {AsyncComponent} from '../utils/async';
+
+const DroppableFileInput = (props) => <AsyncComponent loader={() => import('../utils/file-input').then(c => c.DroppableFileInput)} {...props} />;
+
+const AlertManagerYAMLEditor: React.FC<AlertManagerYAMLEditorProps> = ({obj, onCancel=history.goBack}) => {
+  const secret:K8sResourceKind = obj;
+  const alertManagerYaml = _.get(secret, ['data', 'alertmanager.yaml']);
+  const initErrorMsg = _.isEmpty(alertManagerYaml) ? 'Error: alertmanager.yaml not found in Secret "alertmanager-main", in namespace "openshift-monitoring"' : null;
+
+  const [errorMsg, setErrorMsg] = React.useState(initErrorMsg);
+  const [successMsg, setSuccessMsg] = React.useState();
+  const [inProgress, setInProgress] = React.useState(false);
+  const [yamlStringData, setYamlStringData] = React.useState(!_.isEmpty(alertManagerYaml) ? Base64.decode(alertManagerYaml): '');
+
+  const onYAMLChange = fileData => {
+    setYamlStringData(fileData);
+  };
+
+  const save = e => {
+    e.preventDefault();
+    setInProgress(true);
+    const patch = [{ op: 'replace', path: '/data/alertmanager.yaml', value:  Base64.encode(yamlStringData)}];
+    k8sPatch(SecretModel, secret, patch)
+      .then(newSecret => {
+        setSuccessMsg(`${newSecret.metadata.name} has been updated to version ${newSecret.metadata.resourceVersion}`);
+        setInProgress(false);
+      }, err => {
+        setErrorMsg(err.message);
+        setInProgress(false);
+      });
+  };
+
+  return <div className="co-m-pane__body">
+    <form
+      className="co-m-pane__body-group co-alert-manager-yaml-form co-m-pane__form"
+      onSubmit={save}
+    >
+      <p className="co-alert-manager-yaml__explanation">
+        Update this YAML to configure Routes, Receivers, Groupings and other Alert Manager settings
+      </p>
+      {!_.isEmpty(yamlStringData) && <div className="co-alert-manager-yaml__form-entry-wrapper">
+        <div className="co-alert-manager-yaml__form">
+          <div className="form-group">
+            <DroppableFileInput
+              onChange={onYAMLChange}
```

Comment:
```suggestion
              onChange={setYamlStringData}
```

Then `onYAMLChange` could be removed.

Reply:
thanks, fixed